iceberg/tea_scan.cpp

```diff
@@ -140,6 +145,181 @@ void AddSequenceNumberOrFail(const ManifestFile& manifest, ManifestEntry& entry)
+  int res;
+  std::memcpy(&res, value.data(), sizeof(int));
+  return res;
+}
+
+int YearsToDays(int years) {
+  const int max_days_per_year = 366;
+  Ensure(years <= std::numeric_limits<int>::max() / max_days_per_year &&
+             years >= std::numeric_limits<int>::min() / max_days_per_year,
+         std::string(__PRETTY_FUNCTION__) + ": days can overflow");
+  auto start = std::chrono::sys_days{std::chrono::year(1970) / 1 / 1};
+  auto end = start + std::chrono::years(years);
+  auto duration = end - start;
+  return std::chrono::duration_cast<std::chrono::days>(duration).count();
+}
+
+int MonthsToDays(int months) {
+  const int max_days_per_month = 31;
+  Ensure(months <= std::numeric_limits<int>::max() / max_days_per_month &&
+             months >= std::numeric_limits<int>::min() / max_days_per_month,
+         std::string(__PRETTY_FUNCTION__) + ": days can overflow");
+  auto start = std::chrono::sys_days{std::chrono::year(1970) / 1 / 1};
+  auto end = start + std::chrono::months(months);
+  auto duration = end - start;
+  return std::chrono::duration_cast<std::chrono::days>(duration).count();
+}
+
+int DaysToHours(int days) {
+  const int hours_per_day = 24;
+  Ensure(days <= std::numeric_limits<int>::max() / hours_per_day &&
+             days >= std::numeric_limits<int>::min() / hours_per_day,
+         std::string(__PRETTY_FUNCTION__) + ": hours will overflow");
+  return days * hours_per_day;
+}
+
+int64_t HoursToMicros(int64_t hours) {
+  const int64_t micros_per_hour = 3600ll * 1000ll * 1000ll;
+  Ensure(hours <= std::numeric_limits<int64_t>::max() / micros_per_hour &&
+             hours >= std::numeric_limits<int64_t>::min() / micros_per_hour,
+         std::string(__PRETTY_FUNCTION__) + ": micros will overflow");
+  return hours * micros_per_hour;
+}
+
+class ManifestFileStatsGetter : public filter::IStatsGetter {
+ public:
+  ManifestFileStatsGetter(const std::unordered_map<std::string, TypeID>& name_to_type,
+                          std::unordered_map<std::string, int>&& name_to_position,
+                          const std::vector<PartitionFieldSummary>& partitions, std::shared_ptr<PartitionSpec> spec)
+      : name_to_type_(name_to_type),
+        name_to_position_(std::move(name_to_position)),
+        partitions_(partitions),
+        spec_(spec) {
+    Ensure(spec != nullptr, std::string(__PRETTY_FUNCTION__) + ": spec is nullptr");
+  }
+
+  std::optional<filter::GenericStats> GetStats(const std::string& column_name,
+                                               filter::ValueType value_type) const override {
+    auto it = name_to_position_.find(column_name);
+    if (it == name_to_position_.end()) {
+      return std::nullopt;
+    }
+    const auto position = it->second;
+    Ensure(position >= 0 && position < partitions_.size(), std::string(__PRETTY_FUNCTION__) + ": invalid position");
+
+    const auto ice_type = [&]() {
```

Comment:
Could you avoid lambda. I've read it as you pass lambda to `TypesToStatsConverter`. It's not clear that it's just and inline initialization.

We even could avoid iterators if it's not a bottleneck for clear code. Just
```
Ensure(name_to_position_.contains(column_name) && name_to_type_.contains(column_name));
```